server/public/model/utils.go

```diff
@@ -484,17 +485,41 @@ func ArrayToJSON(objmap []string) string {
 	return string(b)
 }
 
+// Deprecated: ArrayFromJSON is deprecated,
+// use ArrayFromJSONLimited or ArrayFromJsonNonSort instead
 func ArrayFromJSON(data io.Reader) []string {
 	var objmap []string
-
 	json.NewDecoder(data).Decode(&objmap)
 	if objmap == nil {
 		return make([]string, 0)
 	}
-
 	return objmap
 }
 
+func ArrayFromJSONLimited(data io.Reader, maxBytes int64) ([]string, error) {
+	var obj []string
+	lr := io.LimitReader(data, maxBytes)
+	err := json.NewDecoder(lr).Decode(&obj)
+	if err != nil || obj == nil {
+		return nil, err
+	}
+
+	// Remove duplicate IDs as it can bring a significant load to the database.
+	return RemoveDuplicateStrings(obj), nil
+}
+
+func ArrayFromJSONNonSort(data io.Reader, maxBytes int64) ([]string, error) {
```

Comment:
Personally I'd prefer if the default case wasn't sorting vs the negated alternative. And then have a `SortedArrayFromJSON` version. It may also be worth reviewing where we actually need to sort as a potential performance improvement (can be deferred of course).

Reply:
Yeah, the explicit naming works for me. Not expecting to alter any functionality for the time being.